src/openapi-generator.ts

```diff
@@ -655,19 +655,52 @@ export class OpenAPIGenerator {
     zodSchema: ZodObject<ZodRawShape>,
     isNullable: boolean
   ): SchemaObject {
+    const extendedFrom = zodSchema._def.openapi?.extendedFrom;
+
     const propTypes = zodSchema._def.shape();
     const unknownKeysOption = zodSchema._unknownKeys as UnknownKeysParam;
 
     const requiredProperties = Object.entries(propTypes)
       .filter(([_key, type]) => !this.isOptionalSchema(type))
       .map(([key, _type]) => key);
 
+    const properties = mapValues(propTypes, propSchema =>
+      this.generateInnerSchema(propSchema)
+    );
+
+    if (extendedFrom) {
+      const alreadyRegistered = Object.keys(
+        this.schemaRefs[extendedFrom]?.properties ?? {}
+      );
+      const alreadyRequired = this.schemaRefs[extendedFrom]?.required ?? [];
```

Comment:
If the schema does not exist in the registry maybe we should fall back to the "else" case instead of defaulting to empty arrays and letting it flow through the "extendedFrom" logic?

Reply:
I see your point. This code shouldn't actually ever occur to be honest unless someone passes `extendedFrom` manually or the order is messed up somehow. I think I am more inclined to throw an error instead :thinking: .